src/celestia/qt/qttourguide.cpp

```diff
@@ -0,0 +1,110 @@
+// qttourguide.cpp
+//
+// Copyright (C) 2023, the Celestia Development Team
+//
+// Celestia dialog to activate tour guide.
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+
+
+#include "qttourguide.h"
+
+#include <celestia/destination.h>
+#include <celengine/simulation.h>
+#include <celengine/selection.h>
+#include <celestia/celestiacore.h>
+#include <celengine/observer.h>
+
+#include <string>
+
+#include <Eigen/Core>
+
+#include <QDialogButtonBox>
+#include <QPushButton>
+#include <QComboBox>
+#include <QString>
+#include <QLabel>
+#include <QWidget>
+
+const DestinationList* destinations;
+
+
+TourGuideDialog::TourGuideDialog(QWidget *parent, CelestiaCore* _appCore) :
+    QDialog(parent),
+    appCore(_appCore)
+{
+    ui.setupUi(this);
+
+    destinations = appCore->getDestinations();
+    int index = -1;
+    if (destinations != NULL)
+    {
+        for (DestinationList::const_iterator iter = destinations->begin();
```

Comment:
Use ranged for loop

```c++
for (const Destination* dest : destinations)
{
    if (dest != nullptr)
    {
        // ...
    }
}
```

Reply:
better
```c++
for (const Destination* dest : destinations)
{
    if (dest == nullptr)
        continue;
    // ...
}
```

Reply:
For ranged loop will not compile.  Throws errors on conflicting variable types.  Cleaned up some code inside of loop.

Reply:
ah sorry, should be `for (const Destination* dest : *destinations)`

Reply:
Done